src/main/java/net/dries007/tfc/common/blockentities/rotation/WindmillBlockEntity.java

```diff
@@ -69,6 +74,8 @@ public static void clientTick(Level level, BlockPos pos, BlockState state, Windm
             : Math.max(targetSpeed, currentSpeed - LERP_SPEED);
 
         rotation.setSpeed(nextSpeed);
+
+        windmill.hasFullIdenticalSet = windmill.hasFullIdenticalSet();
```

Comment:
Calling this on tick is poor - it's negating most of the impact from caching this in the first place - remove this call + assignment here.